nuget/lib/dependabot/nuget/update_checker/repository_finder.rb

```diff
@@ -315,6 +315,21 @@ def repos_from_config_file(config_file)
           known_urls.include?(s.fetch(:url))
         end
 
+        # filter out based on packageSourceMapping
+        package_mapping_elements = doc.xpath("/configuration/packageSourceMapping/packageSource/package[@pattern]")
+        matching_package_elements = package_mapping_elements.select do |package_element|
+          pattern = package_element.attribute("pattern").value
+          File.fnmatch(pattern, @dependency.name, File::FNM_CASEFOLD) # case insensitive check
```

Comment:
Am I reading it right that this isn't REALLY a file operation, it just uses the `File.fnmatch` method because it uses the same globbing patterns? If so you might leave a comment clarifying that, because I thought this was going out to disk at first.

Reply:
Good call, comment added.